### ISSUE TITLE
Inference for WeightNormalization

### ISSUE THREAD

**Issue by alexmil2019** [tensorflow/tensorflow#47787]
WeightNormalization according to
`https://www.tensorflow.org/addons/api_docs/python/tfa/layers/WeightNormalization` 
creates a wrapper around the kernel for normalization. it uses `tf.nn.l2_normalize` for normalization.
The issue is during inference which you don't need to run `tf.nn.l2_normalize` every time.
Is there any tool or a way to replace the kernel with normalized kernel after training?

**Comment by Saduf2019** [comment]
@alexmil2019 
Please share simple stand alone code to replicate the issue reported or a colab gist.

**Comment by alexmil2019** [comment]
```
def get_model():
  in_shape = (3, 3, 1)
  in_feat = tf.keras.Input(shape=in_shape, name="model_input")
  output = WeightNormalization(tf.keras.layers.Conv2D(filters=4,
                                                    kernel_size=3,
                                                    strides=1,
                                                    padding='VALID',
                                                    data_format='channels_last',
                                                    activation=None,
                                                    kernel_initializer='ones',
                                                      use_bias=True))(in_feat)

  return tf.keras.Model(inputs=[in_feat], outputs=[output])

```

Here is the output of profiler in prediction/inference mode:
```
==================Model Analysis Report======================
Incomplete shape.

Doc:
scope: The nodes in the model graph are organized by their names, which is hierarchical like filesystem.
flops: Number of float operations. Note: Please read the implementation for the math behind it.

Profile:
node name | # float_ops
_TFProfRoot (--/152 flops)
  model/weight_normalization/l2_normalize (36/116 flops)
    model/weight_normalization/l2_normalize/Square (36/36 flops)
    model/weight_normalization/l2_normalize/Sum (32/32 flops)
    model/weight_normalization/l2_normalize/Rsqrt (8/8 flops)
    model/weight_normalization/l2_normalize/Maximum (4/4 flops)
  model/weight_normalization/mul (36/36 flops)

======================End of Report==========================
========================================================
Total Flops : 152
```

**Comment by Saduf2019** [comment]
@alexmil2019 
I ran the code shared and there is no output, please share a colab gist with the issue reported.

**Comment by alexmil2019** [comment]
fixed.
